File: Data Structures & Algorithms/sliding-window-maximum/submission-4.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        q=deque()
        l,r=0,0
        res=[] # store index

        while r<len(nums):
            # before appending the index we should remove all the index which has value less than curr index
            # in this way the max ele in the curr window will always be at first pos of q
            while q and nums[q[-1]] <nums[r]:
                q.pop()
            # now append the curr_index
            q.append(r)

            if q[0]<l:
                q.popleft()
            if(r+1)>=k:
                res.append(nums[q[0]])
                l+=1
            r+=1
        return res


        

File: Data Structures & Algorithms/merge-sorted-array/submission-4.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int currIndex=m+n-1;
        int n1=m-1;
        int n2=n-1;
        while(n1>=0 && n2>=0){
            if(nums1[n1]>nums2[n2]){
                nums1[currIndex]=nums1[n1];
                n1-=1;
            }
            else{
             nums1[currIndex]=nums2[n2];
             n2-=1;
            }
            currIndex-=1;
        }
        while(n1>=0){
            nums1[currIndex]=nums1[n1];
            n1-=1;
            currIndex-=1;
        }
        while(n2>=0){
            nums1[currIndex]=nums2[n2];
            n2-=1;
            currIndex-=1;

        }
    }
}